llvm/lib/Target/DirectX/DXILShaderFlags.h

```diff
@@ -120,16 +127,17 @@ class ShaderFlagsAnalysisPrinter
 /// This is required because the passes that will depend on this are codegen
 /// passes which run through the legacy pass manager.
 class ShaderFlagsAnalysisWrapper : public ModulePass {
-  DXILModuleShaderFlagsInfo MSFI;
+  std::unique_ptr<DXILModuleShaderFlagsInfo> MSFI;
```

Comment:
Out of interest, did you consider `std::optional` for this?

Reply:
> +1
> 
> I also question if we need the optional. Why was the old pattern of the uninitialized flag structure a problem?

Reverted to old pattern; changed to use `initialize()` instead of a constructor to avoid dynamic memory allocation about which concerns were expressed.